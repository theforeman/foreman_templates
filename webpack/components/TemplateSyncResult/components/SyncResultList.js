import React from 'react';
import { ListView } from 'patternfly-react';
import PropTypes from 'prop-types';

import Pagination from 'foremanReact/components/Pagination/PaginationWrapper';

import SyncedTemplate from './SyncedTemplate';
import { templatesPage } from '../TemplateSyncResultHelpers';
import ListViewHeader from './ListViewHeader';

const SyncResultList = ({ pagination, pageChange, templates, editPaths }) => (
  <ListView>
    <ListViewHeader />
    {templatesPage(templates, pagination).map(template => (
      <SyncedTemplate
        template={template}
        key={template.name}
        editPath={editPaths[template.className]}
      />
    ))}
    <Pagination
      viewType="list"
      itemCount={templates.length}
      pagination={pagination}
      onChange={pageChange}
      dropdownButtonId="template-sync-result-dropdown"
    />
  </ListView>
);

SyncResultList.propTypes = {
  pagination: PropTypes.shape({
    page: PropTypes.number,
    perPage: PropTypes.number,
  }).isRequired,
  pageChange: PropTypes.func.isRequired,
  templates: PropTypes.array.isRequired,
  editPaths: PropTypes.object.isRequired,
};

export default SyncResultList;
