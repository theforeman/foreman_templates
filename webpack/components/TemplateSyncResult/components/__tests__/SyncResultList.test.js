import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import SyncResultList from '../SyncResultList';
import { importTemplates } from '../../__fixtures__/templateSyncResult.fixtures';

const noop = () => {};

const editPaths = {
  JobTemplate: '',
  Ptable: '',
  ProvisioningTemplate: '',
};

const fixtures = {
  'should render': {
    pageChange: noop,
    templates: importTemplates,
    editPaths,
    pagination: {
      page: 1,
      perPage: 20,
    },
  },
};

describe('SyncResultList', () =>
  testComponentSnapshotsWithFixtures(SyncResultList, fixtures));
