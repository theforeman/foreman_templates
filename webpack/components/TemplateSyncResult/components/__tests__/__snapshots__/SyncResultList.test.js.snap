// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncResultList should render 1`] = `
<ListView
  className=""
>
  <ListViewHeader />
  <SyncedTemplate
    editPath=""
    key="null"
    template={
      Object {
        "additionalErrors": Array [
          "No \\"name\\" found in metadata",
        ],
        "errors": null,
        "name": null,
        "templateFile": "random_template.erb",
      }
    }
  />
  <SyncedTemplate
    editPath=""
    key="EPEL"
    template={
      Object {
        "additionalErrors": null,
        "className": "ProvisioningTemplate",
        "errors": Object {
          "base": Array [
            "This template is locked",
          ],
        },
        "humanizedClassName": "Provisioning Template",
        "kind": "Provision",
        "locked": true,
        "name": "EPEL",
        "snippet": true,
        "templateFile": "epel.erb",
      }
    }
  />
  <SyncedTemplate
    editPath=""
    key="CoreOS default"
    template={
      Object {
        "additionalErrors": null,
        "className": "Ptable",
        "errors": Object {},
        "humanizedClassName": "Ptable",
        "locked": false,
        "name": "CoreOS default",
        "snippet": false,
        "templateFile": "coreos_default.erb",
      }
    }
  />
  <SyncedTemplate
    editPath=""
    key="null"
    template={
      Object {
        "additionalErrors": "No \\"model\\" found in metadata",
        "errors": null,
        "name": null,
        "templateFile": "no_model.erb",
      }
    }
  />
  <SyncedTemplate
    editPath=""
    key="Kickstart fake"
    template={
      Object {
        "additionalErrors": null,
        "className": "ProvisioningTemplate",
        "errors": Object {
          "name": Array [
            "has already been taken",
          ],
        },
        "humanizedClassName": "Provisioning Template",
        "locked": true,
        "name": "Kickstart fake",
        "snippet": true,
        "templateFile": "kickstart_fake.erb",
      }
    }
  />
  <PaginationWrapper
    dropdownButtonId="template-sync-result-dropdown"
    itemCount={5}
    onChange={[Function]}
    pagination={
      Object {
        "page": 1,
        "perPage": 20,
      }
    }
    viewType="list"
  />
</ListView>
`;
